packages/material-ui/src/styles/styled.js

```diff
@@ -1,14 +1,11 @@
-import { styled as styledWithoutDefault } from '@material-ui/styles';
-import defaultTheme from './defaultTheme';
+import legacy_styled from './legacy_styled';
 
+// @deprecated Follow the upgrade guide on http://next.material-ui.com/guides/migration-v4/#theme
 const styled = (Component) => {
-  const componentCreator = styledWithoutDefault(Component);
-
-  return (style, options) =>
-    componentCreator(style, {
-      defaultTheme,
-      ...options,
-    });
+  console.error(
+    'Material-UI: This method will no longer be available in v5. Please use the legacy_styled utility instead>.',
```

Comment:
How does a developer solve "use the legacy_styled utility instead"?

```suggestion
    'Material-UI: This method will no longer be available in v5. Please use the legacy_styled utility instead.',
```

Reply:
Reverted this changes from the PR